File: analytics-service/src/main/java/com/shiptrackpro/analytics/service/AnalyticsService.java
package com.shiptrackpro.analytics.service;

import com.shiptrackpro.analytics.dto.response.*;

import java.util.List;
import java.util.UUID;

public interface AnalyticsService {

    AdminDashboardDTO getAdminDashboard();

    CustomerDashboardDTO getCustomerDashboard(UUID customerId);

    List<ShipmentVolumeDataPointDTO> getAdminVolumeSeries(String period);

    List<ShipmentVolumeDataPointDTO> getCustomerVolumeSeries(UUID customerId, String period);

    StatusDistributionDTO getStatusDistribution();

    DelayAnalysisDTO getDelayAnalysis();
}

File: frontend/src/services/analyticsService.js
import api from './api';

/**
 * Analytics Service API calls.
 */
const analyticsService = {
  getAdminDashboard: () =>
    api.get('/analytics/admin/dashboard'),

  getCustomerDashboard: () =>
    api.get('/analytics/customer/dashboard'),

  getAdminVolumeSeries: (period = '14DAYS') =>
    api.get('/analytics/admin/shipments/volume', { params: { period } }),

  getCustomerVolumeSeries: (period = '14DAYS') =>
    api.get('/analytics/customer/volume', { params: { period } }),

  getStatusDistribution: () =>
    api.get('/analytics/admin/shipments/status-distribution'),

  getAdminDelays: () =>
    api.get('/analytics/admin/delays'),

  generateReport: (data) =>
    api.post('/analytics/reports/generate', data),

  getMyReports: () =>
    api.get('/analytics/reports/my'),
};

export default analyticsService;
